### DISCUSSION TITLE
Error: No route found, path ends with undefined

### DISCUSSION THREAD

**Discussion by Rikijs** [symfony/symfony#45011] [Q&A] [answered]
Hello!

On the site I am building countdown timer that represents time to session expiration.
Timer works, but the problem occurs when it ends and user is redirected (by JavaScript) to app_logout path!

I supply `app_logout` path by the means of data attribute (`data-path-redirect` which resolves
to "https://127.0.0.1:8000/en/logout">) and script uses it to to call window.location.href;
Unfortunately, i get Symfony error and website is stopped on the page with URL 
`https://127.0.0.1:8000/en/admin/undefined` instead of log out URL `https://127.0.0.1:8000/en/logout`

The error message is: `No route found for "GET https://127.0.0.1:8000/en/admin/undefined" (from "https://127.0.0.1:8000/en/admin/start")`

This is happening on Windows 10 Pro, XAMPP (PHP v7.4.27), with PHP built in dev server. Project is built with Symfony v5.4.2.
On the web i found (https://github.com/sulu/sulu/issues/1925) from the end of 2016 - yet the proposed solution does not work for me...

Have someone else encountered similar error?
Any help appreciated!

**Comment by stof** [comment]
My guess is that your javascript is reading the `data-path-redirect` attribute on the wrong place (or in the wrong way), getting `undefined` rather than `"https://127.0.0.1:8000/en/logout"`. And then, when casting `undefined` to string, it is the `'undefined'` string, which will be resolved as a relative URL when assigning it to `window.location.href`, producing this `undefined` at the end of the URL.

**Comment by Rikijs** [comment, marked as answer]
It was exactly the problem!
Turns out i had typo in code defining data attributes and it prevented correct assigning of `app_logout` path value.
Thank you @stof !
